File: programmers/js/level1/모의고사.js
function solution(answers) {
    var answer = [];
    let count = [0, 0, 0];
    let twoAns = [1, 3, 4, 5];
    let threeAns = [3, 1, 2, 4, 5];
    for (let i = 0; i < answers.length; ++i) {
        // 1번 답 i%5 +1 
        const one = i % 5 + 1;
        // 2번 답 i%2==0 -> 2, i/2 -> index [1,3,4,5]
        let two = -1;
        if (i % 2 == 0) {
            two = 2;
        } else {
            two = twoAns[parseInt(i / 2) % 4];
        }
        // 3번 답 i/2 -> index [3,1,2,4,5]
        const three = threeAns[parseInt(i / 2) % 5];
        if (one == answers[i]) count[0]++;
        if (two == answers[i]) count[1]++;
        if (three == answers[i]) count[2]++;
    }
    let max = 0;
    for (let i = 0; i < count.length; ++i) {
        if (count[i] > max) {
            answer = [i + 1];
            max = count[i];
        } else if (count[i] == max) {
            answer.push(i + 1);
        }
    }
    return answer;
}

console.log(solution([1, 2, 3, 4, 5]));

File: programmers/python/level1/모의고사.py
def solution(answers):
    result = []
    count = [0,0,0]
    answer2 = [1,3,4,5]
    answer3 = [3,1,2,4,5]
    for i, answer in enumerate(answers):
        # 1번 1,2,3,4,5
        if i%5+1 == answer:
            count[0] += 1
        # 2번 짝수 2, 홀수 1,3,4,5
        if i%2==0 and answer == 2:
            count[1] += 1
        elif i%2==1 and answer2[(i//2)%4] == answers[i]:
            count[1] += 1
        # 3번 33 11 22 44 55
        if answer3[(i//2)%5] == answers[i]:
            count[2] += 1
    # 가장 높은 사람 찾기
    max = 0
    for i, c in enumerate(count):
        if c > max:
            max = c
            result = [i+1]
        elif c == max:
            result.append(i+1)

    return sorted(result)

print(solution([1,2,3,4,5]))